exporter/datadogexporter/internal/logs/sender_test.go

```diff
@@ -0,0 +1,212 @@
+					"message":  "log 1",
+					"service":  "server",
+				},
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag2:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 2",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag2:true",
+					"hostname": "hostname",
+					"message":  "log 2",
+					"service":  "server",
+				},
+			}},
+			testFn: func(jsonLogs testutil.JSONLogs, counter int) {
+				switch counter {
+				case 0:
+					assert.True(t, jsonLogs.HasDDTag("tag1:true"))
+					assert.Len(t, jsonLogs, 1)
+				case 1:
+					assert.True(t, jsonLogs.HasDDTag("tag2:true"))
+					assert.Len(t, jsonLogs, 1)
+				default:
+					t.Fail()
+				}
+			},
+			numRequests: 2,
+		},
+		{
+			name: "two-batches",
+			payload: []datadogV2.HTTPLogItem{{
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag1:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 1",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag1:true",
+					"hostname": "hostname",
+					"message":  "log 1",
+					"service":  "server",
+				},
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag1:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 2",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag1:true",
+					"hostname": "hostname",
+					"message":  "log 2",
+					"service":  "server",
+				},
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag2:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 3",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag2:true",
+					"hostname": "hostname",
+					"message":  "log 3",
+					"service":  "server",
+				},
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag2:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 4",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag2:true",
+					"hostname": "hostname",
+					"message":  "log 4",
+					"service":  "server",
+				},
+			}},
+			testFn: func(jsonLogs testutil.JSONLogs, counter int) {
+				switch counter {
+				case 0:
+					assert.True(t, jsonLogs.HasDDTag("tag1:true"))
+					assert.Len(t, jsonLogs, 2)
+				case 1:
+					assert.True(t, jsonLogs.HasDDTag("tag2:true"))
+					assert.Len(t, jsonLogs, 2)
+				default:
+					t.Fail()
+				}
+			},
+			numRequests: 2,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var numCalls int
```

Comment:
```suggestion
			var calls int
```
Won't `calls` be enough? Does it need `num*`?